spec/models/services/data_consolidator_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'spec_helper'
+
+def process_entities_to_cmp(entities)
+  entities.flat_map {|e| e[:entity]['id'] = e[:entity]['id'].first['id']; e[:entity]}
+end
+
+describe Maestrano::Connector::Rails::Services::DataConsolidator do
+  let!(:organization) {create(:organization, uid: 'cld-123')}
+  let(:external_entities) {
+    [
+      {
+        'amount' => 8.93,
+        'type' => 'RECEIVE',
+        'public_note' => '',
+        'transaction_date' => '2018-07-31 12:00:00 +0000',
+        'name' => 'INTEREST PAYMENT',
+        'id' => 'TX8934077941420180731',
+        'account_id' => '4518736'
+      },
+      {
+        'amount' => -26.47,
+        'type' => 'SPEND',
+        'public_note' => 'DISBURSEMENT TO MORTGAGOR',
+        'transaction_date' => '2018-01-16 12:00:00 +0000',
+        'name' => 'DISBURSEMENT TO MORTGAGOR',
+        'id' => '201801160',
+        'account_id' => '4518738'
+      },
+      {
+        'amount' => 25.0,
+        'type' => 'RECEIVE',
+        'public_note' => 'LATE CHARGE PAID',
+        'transaction_date' => '2018-01-12 12:00:00 +0000',
+        'name' => 'LATE CHARGE PAID',
+        'id' => '201801121',
+        'account_id' => '4518738'
+      }
+    ]
+  }
+
+  class BankTransactionMapper
+    extend HashMapper
+    map from('name'), to('name')
+    map from('transaction_date'), to('transaction_date')
+    map from('type'), to('type')
+    map from('public_note'), to('public_note')
+    map from('account_id'), to('account_id')
+    map from('id'), to('id')
+    map from('lines'), to('lines')
+    map from('status'), to('status')
+    map from('amount'), to('amount')
+  end
+
+  class BankTransaction < Maestrano::Connector::Rails::Entity
+    def self.id_from_external_entity_hash(entity)
+      entity['id']
+    end
+
+    def self.connec_entity_name
+      'Transaction'
+    end
+
+    # Entity name in external system
+    def self.external_entity_name
+      'Transaction'
+    end
+
+    def self.object_name_from_external_entity_hash(entity)
+      self.id_from_external_entity_hash(entity)
+    end
+
+    def self.mapper_class
+      BankTransactionMapper
+    end
+  end
+
+  describe '#consolidate_external_entities' do
+    let(:transaction) {BankTransaction.new(organization, nil, nil)}
+    let(:data_consolidator) {described_class.new(organization, transaction, {})}
+
+    context 'entity is not immutable, no entity has been pushed to connec! before' do
+      it 'returns all external entities' do
+        allow(BankTransaction).to receive(:immutable?).and_return(false)
```

Comment:
This should be in the context, eg:
```ruby
context 'entity is mutable' do
  before { allow(BankTransaction).to receive(:immutable?).and_return(false) }
  ...
end
```

This will make it easier to read and is good practice if you have multiple tests (only one in this case)